apps/spikeConverter.cpp

```diff
@@ -29,11 +43,52 @@ int main( int argc, char* argv[] )
     }
 
     lunchbox::Clock clock;
-    const brion::SpikeReport in( brion::URI( argv[1] ), brion::MODE_READ );
+    brion::SpikeReport in( brion::URI( argv[1] ), brion::MODE_READ );
     const float readTime = clock.resetTimef();
 
     brion::SpikeReport out( brion::URI( argv[2] ), brion::MODE_WRITE );
-    out.writeSpikes( in.getSpikes( ));
+
+    if( in.getReadMode() == brion::SpikeReport::STREAM )
+    {
+        // Stream-to-File conversion
+        while( in.getNextSpikeTime() != brion::UNDEFINED_TIMESTAMP )
+        {
+            in.waitUntil( brion::UNDEFINED_TIMESTAMP, STREAM_READ_TIMEOUT_MS );
+            out.writeSpikes( in.getSpikes( ));
+        }
+    }
+    else if( out.getReadMode() == brion::SpikeReport::STREAM )
+    {
+        // File-to-Stream conversion
+
+        // leave time for a Zeq connection to establish
+        lunchbox::sleep( STREAM_SEND_DELAY_MS );
+
+        brion::Spikes outSpikes;
+        unsigned int frameIndex = 0;
+
+        const brion::Spikes& inSpikes = in.getSpikes();
+        for( brion::Spikes::const_iterator spikeIt = inSpikes.begin();
+             spikeIt != inSpikes.end(); ++spikeIt )
+        {
+            if( spikeIt->first >= (frameIndex+1) * STREAM_FRAME_LENGTH_MS )
+            {
+                out.writeSpikes( outSpikes );
+                printSentFrame( frameIndex++, outSpikes.size( ));
+                outSpikes.clear();
+                lunchbox::sleep( STREAM_SEND_FREQ_MS );
+            }
+            outSpikes.insert( *spikeIt );
+        }
+        out.writeSpikes( outSpikes );
+        printSentFrame( frameIndex, outSpikes.size( ));
+    }
+    else
+    {
+        // File-to-File conversion
+        out.writeSpikes( in.getSpikes( ));
+    }
```

Comment:
Agreed on both points, since isStream should be removed from the API, at least for what it is used now.